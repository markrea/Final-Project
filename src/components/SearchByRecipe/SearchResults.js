import React from "react";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const Result = ({ results }) => {
  const calories = Math.round(results.calories);
  return (
    
      <div key={results.image} className="innerSearchResult">
        <img src={results.image} alt="mockAlt" className="card-photo" />
        <figcaption className="caption">
          <b>{results.title}</b>
        </figcaption>
        <figcaption className="caption">
          Ready in {results.readyInMinutes}
          mins
        </figcaption>
        <figcaption className="caption">
          Serves
          {results.serves}
        </figcaption>
        <figcaption className="caption"> {calories} Calories</figcaption>
        <a href={results.link} target="_blank" rel="noopener noreferrer">
          View Recipe
        </a>
      </div>
   
  );
};
const SearchResults = ({ searchResults }) => {
  if (searchResults[0]) {
    return (
      <div className="SearchResults" data-testid="results-div">
        <div className="test">
          {searchResults.map((results) => (
            <Result key={results.id} results={results} />
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="noResults">
      <p>No recipes found. Please try another search</p>
    </div>
  );
};
SearchResults.propTypes = {
  Result: PropTypes.array,
};
export default SearchResults;
