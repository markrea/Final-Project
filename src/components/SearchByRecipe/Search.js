import React, { useState } from "react";
import Proptypes from "prop-types";
import Loader from "react-loader-spinner";
import getRecipes2 from "../../requests/getRecipes2";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

import "../../styles/Search.css";
// eslint-disable-next-line import/no-named-as-default-member
import SearchForm from "./SearchForm";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");
  const [diet, setDiet] = useState("");
  const [intolerances, setIntolerances] = useState([]);
  const [loading, setLoading] = useState(false);

  const handleSubmit = async (event) => {
    setLoading(true);

    event.preventDefault(); // stops browser reloading
    setSearchResults(await getRecipes2(value, diet, intolerances));
    setLoading(false);
  };

  const handleChange = async (e) => {
    setValue(e.target.value);
  };

  const handleSelect = async (e) => {
    setDiet(e.target.value);
  };

  return (
    <div className="searchFormDiv" data-testid="searchFormDiv">
      <SearchForm
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        handleSelect={handleSelect}
        query={value}
        diet={diet}
        setIntolerances={setIntolerances}
      />
      {loading && (
        <Loader
          className="loader"
          type="ThreeDots"
          color="green"
          height={60}
          width={60}
          timeout={3000}
        />
      )}
    </div>
  );
};

Search.propTypes = {
  setSearchResults: Proptypes.objectOf.isRequired,
};
export default Search;
