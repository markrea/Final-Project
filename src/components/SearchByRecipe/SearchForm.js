import React from "react";
import "../../styles/Search.css";
import Select from "react-select";
import makeAnimated from "react-select/animated";

const SearchForm = ({
  handleSubmit,
  handleChange,
  handleSelect,
  query,
  diet,
  setIntolerances,
}) => {
  const options = [
    { value: "dairy", label: "Dairy" },
    { value: "egg", label: "Egg" },
    { value: "peanut", label: "Peanut" },
    { value: "seafood", label: "Seafood" },
    { value: "sesame", label: "Sesame" },
    { value: "shellfish", label: "Shellfish" },
    { value: "soy", label: "Soy" },
    { value: "treenut", label: "Tree Nut" },
  ];
  return (
    <form
      type="text"
      className="input-form"
      onSubmit={handleSubmit}
      data-testid="searchForm"
    >
      <fieldset>
        
        {/* <label htmlFor="diet">Diet</label> */}
        <select
          id="diet"
          name="diet"
          size="1"
          value={diet}          
          placeholder="Select Diet"
          onChange={handleSelect}
        >
          <option defaultValue="null">Select Diet</option>
          <option value="vegetarian">Vegetarian</option>
          <option value="vegan">Vegan</option>
          <option value="glutenfree">Gluten Free</option>
          <option value="pescatarian">Pescastarian</option>
        </select>

        <Select
          isMulti
          autoFocus
          name="colors"
          options={options}
          className="select-container"
          classNamePrefix="select"
          placeholder="Select intolerances"
          onChange={setIntolerances}
          theme={theme => ({
            ...theme,
            borderRadius: "10px",
            backgroundColor: "rgba(196, 208, 162, 1)",
          })}
        />
        <br />
        <input
          id="search"
          name="recipeName"
          type="text"
          placeholder="Enter ingredients"
          onChange={handleChange}
          value={query}
          data-testid="input-id"
        />
        <button type="submit" className="submit">
          Go!
        </button>
        <br />
      </fieldset>
    </form>
  );
};

export default SearchForm;
