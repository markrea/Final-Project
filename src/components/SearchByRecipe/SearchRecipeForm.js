import React, { useState } from "react";
import "../../styles/Search.css";

const SearchRecipeForm = () => {
  const [value, setValue] = useState("");

  const handleSubmit = async (event) => {
    event.preventDefault(); // stops browser reloading
  };

  const handleChange = async (e) => {
    setValue(e.target.value);
  };

  return (
    <form
      type="text"
      className="input-form"
      onSubmit={handleSubmit}
      data-testid="searchRecipeForm"
    >
      <fieldset>
        <input
          id="search"
          name="recipeName"
          type="text"
          placeholder="Enter a recipe"
          onChange={handleChange}
          // value={query}
          data-testid="input-id"
        />
        <button type="submit" className="submit">
          Go!
        </button>
      </fieldset>
    </form>
  );
};

export default SearchRecipeForm;
