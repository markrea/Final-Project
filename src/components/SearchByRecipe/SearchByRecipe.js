import React, { useState } from "react";
import Search from "./Search";
import SearchResultsCard from "./SearchResultsCard";

import "../../styles/SearchByIngredients.css";

const SearchByRecipe = () => {
  const [searchResults, setSearchResults] = useState(null);

  return (
    <div className="SearchByIngredients">
      <div className="TitleText">
        <h4>Search By Recipe</h4>
        <h6>Enter a recipe to get started!</h6>
      </div>
      <Search setSearchResults={setSearchResults} />
      <SearchResultsCard searchResults={searchResults} />
    </div>
  );
};
export default SearchByRecipe;

/* import React, { useState } from "react";
/* import Search from "./Search";
import SearchResultsCard from "./SearchResultsCard";
import SearchForm from "./SearchForm"; 
import SearchForm from "../SearchByRecipe/SearchForm";
import SearchResults from "../SearchByRecipe/SearchResults";
import Search from "../SearchByRecipe/Search";
import SearchResultsCard from "../SearchByRecipe/SearchResultsCard";

import "../../styles/SearchByRecipe.css";

const SearchByRecipe = () => {
  const [searchResults, setSearchResults] = useState(null);
  const [value, setValue] = useState("");

  const handleSubmit =(event) => {
    event.preventDefault(); // stops browser reloading
    //console.log(event);
    //setSearchResults(await getRecipes(value));
  };

  const handleChange =  (e) => {
    //setValue(e.target.value);
    console.log(e.target.value);
  };

  return (
    <div className="SearchByRecipe">
      <div className="SearchByRecipeText">
        <h4>Search By Recipe</h4>
        <h6>Enter a recipe to search</h6>
      </div>
      <SearchForm
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        query={value}
      />
      {/*
      <Search setSearchResults={setSearchResults} />
     <SearchResultsCard searchResults={searchResults} /> 
      <div data-testid="searchRecipeFormDiv">
        <SearchRecipeForm />
        {/*  <SearchResultsCard searchResults={searchResults} /> */}
      </div>
    </div>
  );
};
export default SearchByRecipe;
*/
