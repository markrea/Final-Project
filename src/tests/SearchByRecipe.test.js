import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import SearchByRecipe from "../components/SearchByRecipe/SearchByRecipe";

describe("SearchByRecipe", () => {
  it("renders component correctly", () => {
    const { asFragment } = render(<SearchByRecipe />);
    expect(asFragment).toMatchSnapshot();
  });
});
