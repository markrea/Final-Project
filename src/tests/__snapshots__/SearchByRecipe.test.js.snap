// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByRecipe renders component correctly 1`] = `[Function]`;
